lib/elixir/lib/uri.ex

```diff
@@ -628,7 +628,7 @@ end
 defimpl String.Chars, for: URI do
   def to_string(%{host: host, authority: authority, path: path} = uri)
       when (host != nil or authority != nil) and is_binary(path) and
-             binary_part(path, 0, 1) != "/" do
+             path != "" and binary_part(path, 0, 1) != "/" do
```

Comment:
it is not only about path, but about a host, authority, and others. I have a working version, I can upload to my repo

Reply:
Here's the code
https://github.com/eksperimental/elixir/commit/bff9b98aa164ce227ad2295aa10f2669a914b878

Reply:
this way we avoid checking everywhere down the to_string impelmentation

Reply:
I don't think we should handle both for everything. For example, if you have an empty authority, we are going to output the leading `//`. I don't think we should generally treat empty the same as absent information, unless some specific cases like path.

Reply:
To clarify, the reason we nillify on parsing is exactly so we don't have to handle such cases downstream.